File: PPE/exercicio_01.c
#include <stdio.h>

void troca_1(int *a, int *b);
void troca_2(int a, int b);
void troca_3(int *a, int *b);
void troca_4(int x[]);
int troca_5(int a, int*b);


int main(void)
{
    int a, b;
    printf("Entre com o valor de a: ");
    scanf("%d", &a);
    printf("Entre com o valor de b: ");
    scanf("%d", &b);
    // troca os valores de a e b
    //troca_1(&a, &b); //Ok
    troca_2(a, b); //Não ok
    //troca_3(&a, &b); //Não ok
    //int x[] = {a, b}; 
    //troca_4(x); //Ok
    //a = x[0];
    //b = x[1];
    // = troca_5(a, &b); //Ok
    printf("Trocando os valores de a e b\n");
    printf("a = %d, b = %d\n", a, b);
    

    return 0;
}

void troca_1(int *a, int *b)
{
    int temp = *a;
    *a = *b;
    *b = temp;
}

void troca_2(int a, int b)
{
    int temp = a;
    a = b;
    b = temp;
}

void troca_3(int *a, int *b)
{
    int *temp = a;
    a = b;
    b = temp;
}

void troca_4(int x[])
{
    int temp = x[0];
    x[0] = x[1];
    x[1] = temp;
}

int troca_5(int a, int*b)
{
    int temp = a;
    a = *b;
    *b = temp;
    return a;
}

File: PPE/exercicio_01.py

def main():
    a = 10
    b = 20
    print("a = %d, b = %d\n" % (a, b))
    troca_1(a, b)
    #lista = [a, b]
    #troca_2(lista)
    #a = lista[0]
    #b = lista[1]
    #dupla = (a, b)
    #dupla = troca_3(dupla)
    #a = dupla[0]
    #b = dupla[1]
    #dupla = (a, b)
    #troca_4(dupla)
    #a = dupla[0]
    #b = dupla[1]
    #dic = {'a': a, 'b': b}
    #troca_5(dic)
    #a = dic['a']
    #b = dic['b']
    print("a = %d, b = %d\n" % (a, b))

def troca_1(a, b):
    temp = a
    a = b
    b = temp

def troca_2(lista):
    temp = lista[0]
    lista[0] = lista[1]
    lista[1] = temp


def troca_3(dupla):
    dupla = (dupla[1], dupla[0])
    return dupla

def troca_4(dupla):
    a = dupla[0]
    b = dupla[1]
    temp = a
    a = b
    b = temp
    dupla = (a, b)

def troca_5(dic):
    temp = dic['a']
    dic['a'] = dic['b']
    dic['b'] = temp


def main_():
    a = 10
    b = 20
    print("a = %d, b = %d\n" % (a, b))
    #trocar os valores de a e b
    print("a = %d, b = %d\n" % (a, b))

if __name__ == "__main__":
    main()



